carsus/io/output/base.py

```diff
@@ -33,21 +36,54 @@ class TARDISAtomData:
         Dump all attributes into an HDF5 file
     """
 
-    def __init__(self, gfall_reader, ionization_energies, ions,
+    def __init__(self,
+                 atomic_weights,
+                 ionization_energies,
+                 gfall_reader,
+                 zeta_data,
+                 chianti_reader=None,
                  lines_loggf_threshold=-3,
                  levels_metastable_loggf_threshold=-3):
 
+        # TODO: pass these params to the function as `gfall_params`
         self.levels_lines_param = {
             "levels_metastable_loggf_threshold":
             levels_metastable_loggf_threshold,
             "lines_loggf_threshold": lines_loggf_threshold
         }
 
-        self.ions = parse_selected_species(ions)
+        self.atomic_weights = atomic_weights
+        self.ionization_energies = ionization_energies
+        self.ground_levels = ionization_energies.get_ground_levels()
+
+        gfall_ions = gfall_reader.levels.index.tolist()
```

Comment:
The following code block (until line 63) can be simplified by using pandas methods of the index and the dataframe:
```
gfall_ions = gfall_reader.levels.index.droplevel(level=2).unique()
gfall_ions = gfall_ions.sort_values().to_list()
```
should (hopefully) do the same as the code block and would be simpler to read.